addons/binding/org.openhab.binding.onebusaway/src/main/java/org/openhab/binding/onebusaway/internal/handler/StopHandler.java

```diff
@@ -0,0 +1,273 @@
+import com.google.common.collect.Lists;
+import com.google.common.collect.Multimap;
+import com.google.gson.Gson;
+
+/**
+ * The {@link StopHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Shawn Wilsher - Initial contribution
+ */
+public class StopHandler extends BaseBridgeHandler {
+
+    public static final ThingTypeUID SUPPORTED_THING_TYPE = THING_TYPE_STOP;
+
+    private StopConfiguration config;
+    private Gson gson;
+    private HttpClient httpClient;
+    private Logger logger = LoggerFactory.getLogger(StopHandler.class);
+    private ScheduledFuture<?> pollingJob;
+    private Runnable pollingRunnable = new Runnable() {
+        @Override
+        public void run() {
+            fetchAndUpdateStopData();
+        }
+    };
+    private long routeDataLastUpdateMs = 0;
+    private Multimap<String, ObaStopArrivalResponse.ArrivalAndDeparture> routeData = ArrayListMultimap.create();
+    private List<RouteDataListener> routeDataListeners = new CopyOnWriteArrayList<>();
+
+    public StopHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (RefreshType.REFRESH == command) {
+            logger.debug("Refreshing {}...", channelUID);
+            forceUpdate();
+        } else {
+            logger.warn("The OneBusAway binding is a read-only binding and can not handle commands.");
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleUpdate(ChannelUID channelUID, State newState) {
+        logger.warn("The OneBusAway binding is a read-only binding and can not handle channel updates.");
+        super.handleUpdate(channelUID, newState);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        logger.debug("Initializing OneBusAway stop bridge...");
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Checking configuration...");
+
+        config = loadAndCheckConfiguration();
+        if (config == null) {
+            logger.debug("Initialization of OneBusAway bridge failed!");
+            return;
+        }
+
+        // Do the rest of the work asynchronously because it can take a while.
+        scheduler.submit(new Runnable() {
+            @Override
+            public void run() {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Initializing...");
+                httpClient = new HttpClient();
+                try {
+                    httpClient.start();
+                } catch (Exception e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+                    httpClient = null;
+                    return;
+                }
+                gson = new Gson();
```

Comment:
Ok, same here: Should be fine to do it directly in initialize(), or probably even directly in the constructor (as it does not need to be closed/disposed in any way).